analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionEmptinessCheckingCodeFix.cs

```diff
@@ -0,0 +1,100 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public sealed class CollectionEmptinessCheckingCodeFix : SonarCodeFix
+    {
+        private const string Title = "Use Any() instead";
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CollectionEmptinessChecking.DiagnosticId);
+
+        private readonly CSharpFacade language = CSharpFacade.Instance;
+
+        protected override Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixContext context)
+        {
+            if (root.FindNode(context.Diagnostics.First().Location.SourceSpan)?.FirstAncestorOrSelf<BinaryExpressionSyntax>() is { } binary)
+            {
+                var binaryLeft = binary.Left;
+                var binaryRight = binary.Right;
+
+                if (language.ExpressionNumericConverter.TryGetConstantIntValue(binaryLeft, out var left))
+                {
+                    Simplify(root, binary, binaryRight, language.Syntax.ComparisonKind(binary).Mirror().Compare(left), context);
+                }
+                else if (language.ExpressionNumericConverter.TryGetConstantIntValue(binaryRight, out var right))
+                {
+                    Simplify(root, binary, binaryLeft, language.Syntax.ComparisonKind(binary).Compare(right), context);
+                }
+            }
+            return Task.CompletedTask;
+        }
+
+        public static void Simplify(SyntaxNode root, ExpressionSyntax expression, ExpressionSyntax countExpression, CountComparisonResult comparisonResult, CodeFixContext context) =>
+            context.RegisterCodeFix(
+                CodeAction.Create(Title, c => Replacement(root, expression, (InvocationExpressionSyntax)countExpression, comparisonResult, context)),
+                context.Diagnostics);
+
+        private static Task<Document> Replacement(SyntaxNode root, ExpressionSyntax expression, InvocationExpressionSyntax count, CountComparisonResult comparison, CodeFixContext context)
+        {
+            var any = NotAsExtenion(count)
+                ? AnyAsExtension(count)
+                : AnyFromExtension(count);
```

Comment:
Proposal: To keep the `From` in both cases to describe the difference.
```suggestion
                ? AnyFromStaticMethod(count)
                : AnyFromExtension(count);
```